smart-contracts/contracts/utils/LockProxy.sol

```diff
@@ -0,0 +1,954 @@
+// Sources flattened with hardhat v2.18.3 https://hardhat.org
```

Comment:
Is this just the OZ one? or did we add anything custom?
If not, maybe it's better to use the OZ one?